pkg/controller/deployment/deployment_controller.go

```diff
@@ -556,6 +557,21 @@ func (dc *DeploymentController) getPodMapForDeployment(d *extensions.Deployment,
 	return podMap, nil
 }
 
+// hasEmptyImage returns true if any of the deployment containers have empty image in the container spec.
+func hasEmptyImage(d *extensions.Deployment) bool {
+	for _, container := range d.Spec.Template.Spec.Containers {
+		if len(strings.TrimSpace(container.Image)) == 0 {
```

Comment:
Do we allow empty images in Deployments?

Reply:
@kargakis @tnozicka i believe PodSpec image can be now empty

Reply:
```go
	// Docker image name.
	// More info: https://kubernetes.io/docs/concepts/containers/images
	// This field is optional to allow higher level config management to default or override
	// container images in workload controllers like Deployments and StatefulSets.
	// +optional
	Image string `json:"image,omitempty" protobuf:"bytes,2,opt,name=image"`
```

Reply:
the PodSpec validation have:

```go
// TODO: do not validate leading and trailing whitespace to preserve backward compatibility.
// for example: https://github.com/openshift/origin/issues/14659 image = " " is special token in pod template others may have done similar
if len(ctr.Image) == 0 {
  allErrs = append(allErrs, field.Required(idxPath.Child("image"), ""))
}
```

So apparently `image: " "` should pass

Reply:
yes, space is valid